src/wallet/rpcwallet.cpp

```diff
@@ -639,6 +640,10 @@ UniValue listshieldedunspent(const JSONRPCRequest& request)
             if (hasSaplingSpendingKey) {
                 obj.pushKV("change", pwalletMain->GetSaplingScriptPubKeyMan()->IsNoteSaplingChange(nullifierSet, entry.address, entry.op));
             }
+            const auto& nd = pwalletMain->mapWallet.at(entry.op.hash).mapSaplingNoteData.at(entry.op);
```

Comment:
Would be better to have a getter/find instead of accessing the wallet's map directly from outside of the wallet.